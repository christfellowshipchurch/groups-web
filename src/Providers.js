import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
