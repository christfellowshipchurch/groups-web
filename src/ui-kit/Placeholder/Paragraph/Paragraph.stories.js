import React from 'react';
import { storiesOf } from '@storybook/react';

import Paragraph from '.';

storiesOf('ui-kit/Placeholder/Paragraph', module)
  .add('default', () => <Paragraph />)
  .add('lineNumber', () => <Paragraph lineNumber={20} />)
  .add('firstLineWidth', () => <Paragraph firstLineWidth={'23%'} />)
  .add('lastLineWidth', () => <Paragraph lastLineWidth={'33%'} />)
  .add('showLoadingAnimation', () => <Paragraph showLoadingAnimation />)
  .add('style', () => <Paragraph style={{ background: 'salmon' }} />);
