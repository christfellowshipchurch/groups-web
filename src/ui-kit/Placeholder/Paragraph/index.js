import React from 'react';
import styled from 'styled-components';
import ReactPlaceholder from 'react-placeholder';
import PropTypes from 'prop-types';
import { times } from 'lodash';

import Line from '../Line';

const ParagraphLine = styled(Line).attrs({
  className: 'ui-kit-Placeholder-ParagraphLine',
})`
  height: 1rem;
  /* matches P component styles. line-height - font-size */
  margin: calc(${({ theme }) => theme.helpers.verticalRhythm(1)} - 1rem) 0;
`;

const Paragraph = ({
  isLoading,
  firstLineWidth,
  lastLineWidth,
  lineNumber,
  showLoadingAnimation,
  style,
  width,
  ...props
}) => (
  <ReactPlaceholder
    customPlaceholder={times(lineNumber, (i) => {
      let lineWidth = width;
      if (i === 0 && firstLineWidth) lineWidth = firstLineWidth;
      if (i === lineNumber - 1 && lastLineWidth) lineWidth = lastLineWidth;
      return (
        <ParagraphLine
          key={i}
          width={lineWidth}
          showLoadingAnimation={showLoadingAnimation}
          style={style}
        />
      );
    })}
    ready={!isLoading}
    {...props}
  />
);

Paragraph.defaultProps = {
  isLoading: true,
  lineNumber: 3,
  width: '100%',
  lastLineWidth: '75%',
  firstLineWidth: '85%',
};

Paragraph.propTypes = {
  firstLineWidth: PropTypes.string,
  lastLineWidth: PropTypes.string,
  isLoading: PropTypes.bool,
  lineNumber: PropTypes.number,
  showLoadingAnimation: PropTypes.bool,
  style: PropTypes.any, // eslint-disable-line
  width: PropTypes.string,
};

export default Paragraph;
