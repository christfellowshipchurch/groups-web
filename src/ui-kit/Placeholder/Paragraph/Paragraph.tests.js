import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import Paragraph from '.';

describe('The Paragraph placeholder', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Paragraph />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom lineNumber', () => {
    const tree = renderer.create(
      <Providers>
        <Paragraph lineNumber={50} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom firstLineWidth', () => {
    const tree = renderer.create(
      <Providers>
        <Paragraph firstLineWidth={'23%'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a custom lastLineWidth', () => {
    const tree = renderer.create(
      <Providers>
        <Paragraph lastLineWidth={'33%'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('accepts custom line styles', () => {
    const customStyle = { background: 'salmon' };

    const tree = renderer.create(
      <Providers>
        <Paragraph style={customStyle} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render showLoadingAnimation', () => {
    const tree = renderer.create(
      <Providers>
        <Paragraph showLoadingAnimation />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
