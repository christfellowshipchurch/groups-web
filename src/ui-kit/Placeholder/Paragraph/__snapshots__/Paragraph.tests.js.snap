// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Paragraph placeholder accepts custom line styles 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    style={
      Object {
        "background": "salmon",
      }
    }
    width="85%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    style={
      Object {
        "background": "salmon",
      }
    }
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    style={
      Object {
        "background": "salmon",
      }
    }
    width="75%"
  />,
]
`;

exports[`The Paragraph placeholder should render 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="85%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="75%"
  />,
]
`;

exports[`The Paragraph placeholder should render showLoadingAnimation 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 show-loading-animation c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="85%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 show-loading-animation c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 show-loading-animation c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="75%"
  />,
]
`;

exports[`The Paragraph placeholder should render with a custom firstLineWidth 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 23%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="23%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="75%"
  />,
]
`;

exports[`The Paragraph placeholder should render with a custom lastLineWidth 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="85%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 33%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="33%"
  />,
]
`;

exports[`The Paragraph placeholder should render with a custom lineNumber 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="85%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="100%"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: 0.7rem 0;
}

<mark
    className="c0 c1 ui-kit.Placeholder.ParagraphLine ui-kit.Placeholder.Line custom-placeholder"
    width="75%"
  />,
]
`;
